### PR TITLE
Reflogs

### PR DESCRIPTION
**Pull request by OrangeOranges**
## Description

Enables reflogs and adds sno reflog command to display them.
Thanks to Andrew for finding the core.logAllRefUpdates="always" setting, which magically enables writing of reflogs.

## Checklist:

- [X] Have you reviewed your own change?
- [X] Have you included test(s)?
- [ ] Have you updated the [changelog](https://github.com/koordinates/sno/blob/master/CHANGELOG.md)?

### CODE REVIEW

**Comment by OrangeOranges** [comment]
Oh no all the checks failed...

**Review by craigds — COMMENTED** [review]
This looks great. much simpler than I thought, nice work 👍 

Fixes #130 

I checked the test failures and was able to reproduce the failure on my laptop:

```
$ pytest -k test_e2e
...
E                 At index 4 diff: 'HEAD@{4}: ' != 'HEAD@{4}: fast-import'
E                 Full diff:
E                   [
E                    'HEAD@{0}: commit (merge): merge-1',
E                    'HEAD@{1}: checkout: moving from edit-1 to master',
E                    'HEAD@{2}: commit: commit-1',
E                    'HEAD@{3}: checkout: moving from master to edit-1',
E                 -  'HEAD@{4}: fast-import',
E                 +  'HEAD@{4}: ',
```

File: sno/init.py [hunk 1]
```diff
@@ -343,6 +343,9 @@ def init(
     write_repo_version_config(repo, repo_version)
     WorkingCopy.write_config(repo, wc_path, bare)
 
+    # Enable writing to reflogs
+    repo.config["core.logAllRefUpdates"] = "always"
```
Comment (rcoup) [inline sno/init.py:347]: so this automatically persists/writes into the repo `.git/config`? #til

Comment (OrangeOranges) [reply sno/init.py:347]: Yes it writes into the config file.

File: tests/test_branch.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def test_branches(
         r = cli_runner.invoke(["checkout", "HEAD~1"])
         assert r.exit_code == 0, r
 
-        assert text_branches(cli_runner) == ["* (no branch)", "  master"]
+        assert text_branches(cli_runner) == ["* (HEAD detached at 63a9492)", "  master"]
```
Comment (rcoup) [inline tests/test_branch.py:54]: so this changes because git reads it back from the reflog? 🤯 👍

**Review by rcoup — APPROVED** [review]
Nice work 👏

**Comment by rcoup** [comment]
@OrangeOranges GTG, but lets get CI green before merging.

https://github.com/koordinates/sno/pull/253 should sort that, then we can rebase this one and it should be green

File: tests/test_branch.py (hunk 2 above)
Comment (craigds) [reply tests/test_branch.py:54]: it's inexplicable to me, but yes git seems to do this using the reflog... somehow.

**Pull request merged**